docs/plugins/testing_workshop_docs/Article-4-test-coverage.md

```diff
@@ -0,0 +1,104 @@
+* Article 3: [Readers and fixtures](./Article-3-readers-and-fixtures)  
+* Article 4: This article (Test coverage)   
+* Resource links: [testing resources](./Testing-resources.md)   
+
+### This article covers:   
+* [Coverage](#coverage)
+* [pytest --cov](#pytest---cov)    
+ 
+#### Resources  
+The example plugin and all the tests discussed in this article are available in [this GitHub repository](https://github.com/DragaDoncila/plugin-tests).    
+
+## Coverage  
+How do we know when we have tested everything? 
+
+This is where _coverage_ comes in. `pytest-cov` can find out what the coverage of our tests is. Install using 
+`pip install pytest-cov`. This feature provides coverage data for pytest. Coverage is the lines of code that were executed when the tests were run. If there are lines of code that didn’t run at all, they represent a code path we didn’t test, which could hold potential bugs. Coverage gives you an idea of other tests that are needed. 
+
+## pytest --cov  
+
+To run tests with coverage, run `pytest` and add `--cov` pointing to the module you're covering. There is also an option to generate an html report, which we do in this case. 
+
+    (napari-env) user@directory % 
+    pytest --cov=plugin_tests --cov-report=html .  
+
+Note that the `.` (period character) is part of the command.  
+
+    =========================== test session starts ==========================  
+    platform darwin -- Python 3.9.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0  
+    PyQt 5.15.6 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2  
+    rootdir: qt-4.0.2, napari-plugin-engine-0.2.0, napari-0.4.13, cov-3.0.0  
+    collected 2 items  
+ 
+    src/plugin_tests/_tests/test-reader.py ..                           [100%]  
+ 
+    --------- coverage: platform darwin, python 3.0.7-final-0 --------  
+    Coverage HTML written to dir htmlcov  
+ 
+    =========================== 2 passed in 6.72s ==========================  
+
+This command runs the tests, gets the coverage, and then writes the Coverage HTML report to the `htmlcov` directory.
+
+There is a large folder (`htmlcov`) in the directory where the tests were run (`plugin_tests`). 
+
+`(napari-env) plugin-tests $ tree -L 1`  
+`.`  
+`├── LICENSE  `  
+`├── MANIFEST.in`  
+`├── README.md`  
+`├── _pycache__`  
+`├── example_func.py`  
+`├── htmlcov		        # <<=============== directory created by pytest-cov`
+`├── requirements.txt`  
+`├── setup.cfg`  
+`├── setup.py`  
+`├── src`  
+`└── tox.ini`  
+
+
+If we drag the `index.html` file from the list of files in the left panel (to the left of line 32) into a browser, it opens the coverage report. 
+
+![htmlcov directory](../../images/Test_Coverage_htmlcov_directory.png)
+
+We technically have 100% coverage of `test_widget.py` (the fourth line up from Total) because we didn't write a widget. Since there were zero lines of code in the widget, zero were missed, so we had 100% coverage.  
```

Comment:
I would just remove this sentence, as I think it's more likely to confuse people than to clarify anything.